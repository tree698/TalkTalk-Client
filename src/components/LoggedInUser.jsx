import React from 'react';
import Avatar from './Avatar';

export default function LoggedInUser({
  photo,
  username,
  styleForAvatar,
  styleForUsername,
  styleForLoggedInUser,
}) {
  return (
    <div className={styleForLoggedInUser}>
      <Avatar
        photo={photo}
        username={username}
        styleForAvatar={styleForAvatar}
      />
      <span className={styleForUsername}>{username}</span>
    </div>
  );
}
