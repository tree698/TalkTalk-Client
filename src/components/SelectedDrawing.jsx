import React from 'react';
import { useLocation } from 'react-router-dom';
import LoggedInUser from './LoggedInUser';
import { MdOutlineTitle } from 'react-icons/md';
import { BsBrush } from 'react-icons/bs';
import { BsChatLeftDots } from 'react-icons/bs';

export default function SelectedDrawing() {
  const STYLE_FOR_LOGGEDINUSER =
    'flex items-center justify-center shrink-0 my-8';
  const STYLE_FOR_AVATAR = 'w-16 h-16 rounded-full mr-4';
  const STYLE_FOR_USERNAME = 'hidden md:block text-2xl font-semibold';
  const STYLE_FOR_LI = 'flex items-center gap-4 mb-3';

  const {
    state: { fileName, title, username, brush, description, photo },
  } = useLocation();

  return (
    <div className="basis-1/2 flex-1 flex flex-col justify-center items-center">
      <img
        src={`${process.env.REACT_APP_BASE_URL}/uploaded_images/${fileName}`}
        alt={title}
        className="border border-superLightGray shadow-lg rounded-md"
      />
      <LoggedInUser
        photo={photo}
        username={username}
        styleForAvatar={STYLE_FOR_AVATAR}
        styleForUsername={STYLE_FOR_USERNAME}
        styleForLoggedInUser={STYLE_FOR_LOGGEDINUSER}
      />
      <ul className="w-full text-xl px-5 pb-5">
        <li className={STYLE_FOR_LI}>
          <MdOutlineTitle className="text-2xl" />:<span> {title}</span>
        </li>
        <li className={STYLE_FOR_LI}>
          <BsBrush className="text-2xl" />:<span> {brush}</span>
        </li>
        <li className={STYLE_FOR_LI}>
          <BsChatLeftDots className="text-2xl" />:<span> {description}</span>
        </li>
      </ul>
    </div>
  );
}
