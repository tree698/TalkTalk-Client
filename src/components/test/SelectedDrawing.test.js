import { Route } from 'react-router-dom';
import { withRouter } from '../../test/utils';
import SelectedDrawing from '../SelectedDrawing';
import renderer from 'react-test-renderer';
import LoggedInUser from '../LoggedInUser';

jest.mock('../LoggedInUser');

describe('SelectedDrawing', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      withRouter(<Route path="/" element={<SelectedDrawing />} />, {
        state: {
          fileName: 'test',
          title: 'test',
          username: 'test',
          brush: 'test',
          description: 'test',
          photo: 'test',
        },
      })
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
