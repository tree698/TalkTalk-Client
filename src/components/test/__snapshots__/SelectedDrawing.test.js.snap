// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedDrawing renders correctly 1`] = `
<div
  className="basis-1/2 flex flex-col justify-center items-center px-10 md:px-12 lg:pl-0 lg:pr-8"
>
  <img
    alt="test"
    className="w-full border border-superLightGray shadow-lg rounded-md"
    src="test"
  />
  <ul
    className="w-full text-[10px] md:text-xs lg:text-sm"
  >
    <li
      className="flex items-center w-full gap-x-1 md:gap-x-2 lg:gap-x-3 mb-1 md:mb-2 "
    >
      <svg
        className="text-sm md:text-base lg:text-lg"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M5 4v3h5.5v12h3V7H19V4H5z"
        />
      </svg>
      :
      <span>
         
        test
      </span>
    </li>
    <li
      className="flex items-center w-full gap-x-1 md:gap-x-2 lg:gap-x-3 mb-1 md:mb-2 "
    >
      <svg
        className="text-sm md:text-base lg:text-lg"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.825.12a.5.5 0 0 1 .132.584c-1.53 3.43-4.743 8.17-7.095 10.64a6.067 6.067 0 0 1-2.373 1.534c-.018.227-.06.538-.16.868-.201.659-.667 1.479-1.708 1.74a8.118 8.118 0 0 1-3.078.132 3.659 3.659 0 0 1-.562-.135 1.382 1.382 0 0 1-.466-.247.714.714 0 0 1-.204-.288.622.622 0 0 1 .004-.443c.095-.245.316-.38.461-.452.394-.197.625-.453.867-.826.095-.144.184-.297.287-.472l.117-.198c.151-.255.326-.54.546-.848.528-.739 1.201-.925 1.746-.896.126.007.243.025.348.048.062-.172.142-.38.238-.608.261-.619.658-1.419 1.187-2.069 2.176-2.67 6.18-6.206 9.117-8.104a.5.5 0 0 1 .596.04zM4.705 11.912a1.23 1.23 0 0 0-.419-.1c-.246-.013-.573.05-.879.479-.197.275-.355.532-.5.777l-.105.177c-.106.181-.213.362-.32.528a3.39 3.39 0 0 1-.76.861c.69.112 1.736.111 2.657-.12.559-.139.843-.569.993-1.06a3.122 3.122 0 0 0 .126-.75l-.793-.792zm1.44.026c.12-.04.277-.1.458-.183a5.068 5.068 0 0 0 1.535-1.1c1.9-1.996 4.412-5.57 6.052-8.631-2.59 1.927-5.566 4.66-7.302 6.792-.442.543-.795 1.243-1.042 1.826-.121.288-.214.54-.275.72v.001l.575.575zm-4.973 3.04.007-.005a.031.031 0 0 1-.007.004zm3.582-3.043.002.001h-.002z"
        />
      </svg>
      :
      <span>
         
        test
      </span>
    </li>
    <li
      className="flex items-center w-full gap-x-1 md:gap-x-2 lg:gap-x-3 mb-1 md:mb-2 "
    >
      <svg
        className="text-lg md:text-base lg:text-xl"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 1a1 1 0 0 1 1 1v8a1 1 0 0 1-1 1H4.414A2 2 0 0 0 3 11.586l-2 2V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12.793a.5.5 0 0 0 .854.353l2.853-2.853A1 1 0 0 1 4.414 12H14a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
        />
        <path
          d="M5 6a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm4 0a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm4 0a1 1 0 1 1-2 0 1 1 0 0 1 2 0z"
        />
      </svg>
      :
      <span>
         
        test
      </span>
    </li>
  </ul>
</div>
`;
